FirebaseAdmin/FirebaseAdmin/Auth/UserRecord.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright 2019, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+
+namespace FirebaseAdmin.Auth
+{
+    /// <summary>
+    /// Contains metadata associated with a Firebase user account. Instances
+    /// of this class are immutable and thread safe.
+    /// </summary>
+    public class UserRecord
+    {
+        private string _uid;
+        private IReadOnlyDictionary<string, object> _customClaims;
+
+        /// <summary>
+        /// The user ID of this user.
+        /// </summary>
+        [JsonProperty("localId")]
+        public string Uid
+        {
+            get => _uid;
+            set
+            {
+                CheckUid(value);
+                _uid = value;
+            }
+        }
+
+        /// <summary>
+        /// Returns custom claims set on this user.
+        /// </summary>
+        [JsonProperty("customAttributes")]
```

Comment:
Backend server accepts this field as a serialized json string. That needs to be handled somewhere here. I'd suggest mark this with JsonIgnore, and have an internal string property:

```
[JsonProperty("customAttributes")]
internal String CustomClaimsString
{
  get
  {
    return serialize(CustomClaims)
  }
}
```